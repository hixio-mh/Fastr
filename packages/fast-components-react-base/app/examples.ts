import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export {LabelExamples};

import TextField from "../src/text-field/examples.data";
export { TextField };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
