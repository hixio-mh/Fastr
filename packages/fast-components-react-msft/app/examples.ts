import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };
