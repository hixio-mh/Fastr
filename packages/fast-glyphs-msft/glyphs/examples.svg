<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>