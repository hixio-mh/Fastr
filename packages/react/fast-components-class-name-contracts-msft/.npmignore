# Source files
src/

# Configs
tsconfig.json