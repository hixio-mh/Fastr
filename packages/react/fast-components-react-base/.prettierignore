coverage/*
dist/*
www/*
src/**/.tmp/