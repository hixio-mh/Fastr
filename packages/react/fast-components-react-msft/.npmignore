# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
www/
build/
coverage/
src/
assets/

# config
babel.config.js
tsconfig.json
webpack.config.js