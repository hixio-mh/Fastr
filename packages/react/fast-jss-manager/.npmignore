# Source files
src/

# config
tsconfig.json