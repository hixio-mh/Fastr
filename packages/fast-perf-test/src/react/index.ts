import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
