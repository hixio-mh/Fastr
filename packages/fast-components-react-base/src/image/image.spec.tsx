import * as React from "react";
import * as ShallowRenderer from "react-test-renderer/shallow";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import Image, {
    ImageClassNameContract,
    ImageHandledProps,
    ImageManagedClasses,
    ImageProps,
    ImageSlot,
    ImageUnhandledProps,
} from "./image";

/*
 * Configure Enzyme
 */
configure({ adapter: new Adapter() });

describe("image", (): void => {
    const managedClasses: ImageClassNameContract = {
        image: "image-class",
        image__picture: "picture-class",
    };
    const alt: string = "Image alt text test string";

    test("should have a displayName that matches the component name", () => {
        expect((Image as any).name).toBe(Image.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(() => {
            shallow(<Image alt="alt" />);
            shallow(
                <Image alt="alt">
                    <source slot={ImageSlot.source} />
                </Image>
            );
        }).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: ImageHandledProps = {
            managedClasses,
            alt,
            src: "https://placehold.it/200x200",
        };
        const unhandledProps: ImageUnhandledProps = {
            "aria-hidden": true,
        };
        const props: ImageProps = { ...handledProps, ...unhandledProps };
        const rendered: any = shallow(<Image {...props} />);

        expect(rendered.prop("aria-hidden")).not.toBe(undefined);
        expect(rendered.prop("aria-hidden")).toEqual(true);
    });

    test("should render an `<img />` element if `src` prop is passed", () => {
        const rendered: any = shallow(
            <Image
                managedClasses={managedClasses}
                alt={alt}
                src={"https://placehold.it/20x20"}
            />
        );

        expect(rendered.type()).toBe("img");
    });

    test("should render a `<picture>` element if children with the prop of `slot='source'` are passed", () => {
        const rendered: any = shallow(
            <Image managedClasses={managedClasses} alt={alt}>
                <source slot={ImageSlot.source} />
            </Image>
        );

        expect(rendered.type()).toBe("picture");
    });

    test("should NOT render with a srcset value if no `srcSet` prop is passed", () => {
        const rendered: any = shallow(
            <Image
                managedClasses={managedClasses}
                alt={alt}
                src={"https://placehold.it/20x20"}
            />
        );

        expect(rendered.prop("srcSet")).toBe(null);
    });

    test("should render with a srcset value when `srcSet` prop is passed", () => {
        const rendered: any = shallow(
            <Image
                managedClasses={managedClasses}
                alt={alt}
                src={"https://placehold.it/20x20"}
                srcSet={
                    "https://placehold.it/20x20/ 767w, https://placehold.it/40x40/ 1w"
                }
            />
        );

        expect(rendered.prop("srcSet")).toBe(
            "https://placehold.it/20x20/ 767w, https://placehold.it/40x40/ 1w"
        );
    });

    test("should NOT render with a sizes value if no `sizes` prop is passed", () => {
        const rendered: any = shallow(
            <Image
                managedClasses={managedClasses}
                alt={alt}
                src={"https://placehold.it/20x20"}
            />
        );

        expect(rendered.prop("sizes")).toBe(null);
    });

    test("should render with a sizes value when `sizes` prop is passed", () => {
        const rendered: any = shallow(
            <Image
                managedClasses={managedClasses}
                alt={alt}
                src={"https://placehold.it/400x400"}
                sizes={"100vw"}
            />
        );

        expect(rendered.prop("sizes")).toBe("100vw");
    });

    // parametrized image class name tests
    [
        {
            name: "should correctly assign empty className",
            imageHandledProps: {} as ImageHandledProps,
            className: "",
            expectedClassName: null,
        },
        {
            name: "should correctly assign className with root className",
            imageHandledProps: {} as ImageHandledProps,
            className: "class-name",
            expectedClassName: "class-name",
        },
        {
            name: "should correctly assign className with root className and input props",
            imageHandledProps: {
                managedClasses: {
                    image: "image-class",
                    image__picture: "picture-class",
                },
            } as ImageHandledProps,
            className: "class-name",
            expectedClassName: "image-class picture-class class-name",
        },
    ].forEach(({ name, imageHandledProps, className, expectedClassName }: any) => {
        test(name, () => {
            const props: ImageProps = { ...imageHandledProps };
            const rendered: any = shallow(
                <Image {...props} className={className}>
                    <source slot={ImageSlot.source} />
                </Image>
            );

            expect(rendered.prop("className")).toEqual(expectedClassName);
        });
    });
});
