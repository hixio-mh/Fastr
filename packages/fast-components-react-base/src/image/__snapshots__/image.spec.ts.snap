// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image Image: 0 1`] = `
<picture
  className="picture"
>
  <source
    media="(min-width: 1779px)"
    srcSet="https://placehold.it/2048x600/2F2F2F/171717"
  />
  <source
    media="(min-width: 1400px)"
    srcSet="https://placehold.it/1778x600/2F2F2F/171717"
  />
  <source
    media="(min-width: 1084px)"
    srcSet="https://placehold.it/1399x600/2F2F2F/171717"
  />
  <source
    media="(min-width: 768px)"
    srcSet="https://placehold.it/1083x500/2F2F2F/171717"
  />
  <source
    media="(min-width: 540px)"
    srcSet="https://placehold.it/767x400/2F2F2F/171717"
  />
  <source
    media="(min-width: 0)"
    srcSet="https://placehold.it/539x300/2F2F2F/171717"
  />
  <img
    alt="Placeholder with grey background and dimension watermark without any imagery"
    className="image image-round"
    src="https://placehold.it/539x300/2F2F2F/171717"
  />
</picture>
`;
